backend/internal/handlers/articles_test.go

```
@@ -0,0 +1,131 @@
+	RegisterArticles(app, hCtx)
+
+	validRequests := []string{
+		"/articles/501",
+		"/articles/Distributed Systems/Raft Consensus.md",
+		"/articles/Distributed%20Systems/Raft%20Consensus.md",
+	}
+
+	for _, reqPath := range validRequests {
+		t.Run(reqPath, func(t *testing.T) {
+			req := makeTestRequest("GET", reqPath)
+			resp, err := app.Test(req, 5000)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)